### PR TITLE
feat(alerts): Add related transactions to the metric alert rules redesign page

### PR DESCRIPTION
**Pull request by taylangocmen**
This adds a `Slowest transactions` section to the alert rule details page for when the data source on the metric alert rule is transactions. It is the counter part to the `Related Issues` section for when the data source is events.

See the [designs](https://www.figma.com/file/w9DyrFKx9lHjjeO0lJ968o/Brainstorm%3A-Alerts-Redesign?node-id=376%3A1827)
Part of the [WOR-498](https://getsentry.atlassian.net/browse/WOR-498)

### CODE REVIEW

File: src/sentry/static/sentry/app/views/alerts/rules/details/body.tsx [hunk 1]
```diff
@@ -300,6 +302,19 @@ export default class DetailsBody extends React.Component<Props> {
                         filter={queryWithTypeFilter}
                       />
                     )}
+                    {rule?.dataset === Dataset.TRANSACTIONS && (
+                      <RelatedTransactions
+                        organization={organization}
+                        location={location}
+                        rule={rule}
+                        projects={((projects as Project[]) || []).filter(project =>
```
Comment (davidenwang) [inline src/sentry/static/sentry/app/views/alerts/rules/details/body.tsx:310]: we might not need to filter by the rule's projects as I try to get only the project slugs relevant to the alert rule. I know there were problems before with that but hopefully should be fixed by now.

**Review by davidenwang — APPROVED** [review]
lgtm

File: src/sentry/static/sentry/app/views/alerts/rules/details/relatedTransactions.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+}
+
+type TableProps = {
+  eventView: EventView;
+  organization: Organization;
+  location: Location;
+  summaryConditions: string;
+
+  projects: Project[];
+  columnTitles?: string[];
```
[the remark this answers is not in the document]
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/alerts/rules/details/relatedTransactions.tsx:44]: 👍

File: src/sentry/static/sentry/app/views/alerts/rules/details/relatedTransactions.tsx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    tableMeta: TableData['meta'],
+    column: TableColumn<keyof TableDataRow>,
+    title: React.ReactNode
+  ): React.ReactNode {
+    const {eventView, location} = this.props;
+
+    const align = fieldAlignment(column.name, column.type, tableMeta);
+    const field = {field: column.name, width: column.width};
+
+    function generateSortLink(): LocationDescriptorObject | undefined {
```
[the remark this answers is not in the document]
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/alerts/rules/details/relatedTransactions.tsx:106]: I'll remove this and replace with a cleaner header cell

File: src/sentry/static/sentry/app/views/alerts/rules/details/relatedTransactions.tsx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+  render() {
+    const {eventView, organization, location} = this.props;
+
+    const {widths} = this.state;
+    const columnOrder = eventView
+      .getColumns()
+      // remove key_transactions from the column order as we'll be rendering it
+      // via a prepended column
+      .filter((col: TableColumn<React.ReactText>) => col.name !== 'key_transaction')
```
[the remark this answers is not in the document]
Comment (taylangocmen) [reply src/sentry/static/sentry/app/views/alerts/rules/details/relatedTransactions.tsx]: Should be fixed I removed this stuff

**Pull request merged**